shell/platform/android/io/flutter/app/FlutterActivityDelegate.java

```diff
@@ -141,7 +141,11 @@ public void onCreate(Bundle savedInstanceState) {
       Window window = activity.getWindow();
       window.addFlags(LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
       window.setStatusBarColor(0x40000000);
-      window.getDecorView().setSystemUiVisibility(PlatformPlugin.DEFAULT_SYSTEM_UI);
+      if (Build.VERSION.SDK_INT >= 30) {
+        window.setDecorFitsSystemWindows(false);
```

Comment:
Should we use https://developer.android.com/reference/androidx/core/view/WindowCompat instead?

Reply:
Is there an advantage of using this? The [window.setDecorFitsSystemWindows(boolean)](https://developer.android.com/reference/android/view/Window#setDecorFitsSystemWindows(boolean)) behavior seems to mirror that of the [windowCompat.setDecorFitsSystemWindows(window, boolean)](https://developer.android.com/reference/androidx/core/view/WindowCompat#setDecorFitsSystemWindows(android.view.Window,%20boolean)) behavior.

Reply:
`{ClassName}Compat` is part of the AndroidX library. It provides a compatibility layer that handles all the inconsistencies between each version behind-the-scene.

[Example 1](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:core/core/src/main/java/androidx/core/view/WindowCompat.java;l=115-122;drc=3f4c39ca3c10360ca232be03a80f31a725068f12)

[Example 2](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:core/core/src/main/java/androidx/core/view/WindowInsetsControllerCompat.java;l=96-107;drc=4b9ea4a8a3e548875e5a5a13f8a7a61ceea58746)

Since the implementation details are abstracted from us, we can worry less about adding if statements for specific versions on our side and focus more on the feature work. In this case, we should try to migrate the deprecated code to use WindowCompat and WindowInsetsControllerCompat instead.